packages/client/src/rtc/publisher.ts

```diff
@@ -244,12 +259,19 @@ export class Publisher {
   private onNegotiationNeeded = async () => {
     console.log('AAA onNegotiationNeeded');
     const offer = await this.publisher.createOffer();
-    if (this.isDtxEnabled && offer.sdp) {
-      offer.sdp = offer.sdp.replace(
-        'useinbandfec=1',
-        'useinbandfec=1;usedtx=1',
-      );
+    let sdp = offer.sdp;
+    if (sdp) {
+      if (this.isDtxEnabled) {
+        sdp = enableDtx(sdp);
+      }
+      if (isReactNative()) {
+        if (this.reactNativePlatform === 'android') {
+          sdp = setPreferredCodec(sdp, 'video', 'vp8');
+        }
+        sdp = setPreferredCodec(sdp, 'audio', this.preferredAudioCodec);
+      }
```

Comment:
This

Reply:
@santhoshvai and @oliverlaz let's make it more straightforward to our integrators and introduce `preferredVideoCodec` over `reactNativePlatform`

Reply:
i.e. 
```
const videoClient = useCreateStreamVideoClient({
    user: USER,
    tokenOrProvider: USER?.custom?.token,
    apiKey: STREAM_API_KEY,
   preferredVideoCodec: Platform.os === "android" && "vp8" 
});
```

Reply:
I think `preferredVideo/AudioCodec` was introduced for "test" purposes. Ideally, the clients should negotiate which codec to use.

In this particular example, Android would announce it only supports VP8. All other clients should switch to VP8 as part of the negotiation process.
cc: @thesyncim @su225 @szuperaz

Reply:
will refactor this 👍